lib/Basic/SwiftRuntime.cpp

```diff
@@ -0,0 +1,57 @@
+//===--- SwiftRuntime.cpp - Swift Runtime Handling --------------*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#include "swift/Badic/SwiftRuntime.h"
+
+using namespace swift;
+
+bool SwiftRuntime::tryParse(StringRef S) {
+  std::size_t dash = S.rfind('-');
+
+  if (dash != StringRef::npos && dash + 1 != S.size() &&
+      (S[dash + 1] < '0' || S[dash + 1] > '9'))
+    dash = StringRef::npos;
+
+  Version = VersionTuple(0);
+
+  StringRef runtime = S.substr(0, dash);
+  if (runtime == "swift") {
+  } else {
+    return true;
+  }
```

Comment:
Its the same style as objc (swift-[version]).

Reply:
Right, I'm saying that the string component is unnecessary because we don't expect to support multiple *kinds* of runtime.  Clang needs to support the legacy runtime, the modern runtime, the GNU runtime, and ObjCFW.  Even if we someday add support for that, it'll still be okay to privilege the standard runtime as "X.Y" instead of "swift-X.Y".